src/quo2/components/community/token_gating.cljs

```diff
@@ -0,0 +1,172 @@
+                   token-requirement-list)]]))
+
+(defn is-community-locked? [community]
+  (let [{gates :gates} community]
+    (or
+     (when (some? gates)
+       (when (vector? (:join gates))
+         (not (is-token-requirement-met? (:join gates)))))
+     false)))
+
+(defn is-channel-locked? [channel]
+  (let [{gates :gates} channel]
+    (or
+     (when (some? gates)
+       (if (vector? (:read gates))
+         (not (is-token-requirement-met? (:read gates)))
+         (when (vector? (:write gates))
+           (not (is-token-requirement-met? (:write gates))))))
+     false)))
+
+(defn token-gating [props]
+  (let [dark?        (theme/dark?)
+        {:keys [community channel]} props
+        type (if (some? community) :community :channel)
+        inner-props (if (= type :community) community channel)
+        name (:name inner-props)
+        emoji (:emoji inner-props)
+        emoji-background-color (:emoji-background-color inner-props)
+        on-enter-channel (:on-enter-channel inner-props)
+        community-avatar (:community-avatar inner-props)
+        gates (:gates inner-props)
+        locked? (if
+                 (= type :community)
+                  (is-community-locked? inner-props)
+                  (is-channel-locked? inner-props))]
+
+    [rn/view {:style (merge (get styles :container) {:background-color (if dark? colors/neutral-90 colors/white)})}
+     [rn/view {:style (merge
+                       (get styles :header-container)
+                       (if
+                        (= type :community)
+                         (get styles :header-spacing-community)
+                         (get styles :header-spacing-channel)))}
+
+      [rn/view {:style (get styles :header-avatar)}
+       (if (= type :community)
+         [rn/image {:source community-avatar
+                    :style (get styles :header-community-avatar-image)}]
+
+         [channel-avatar/channel-avatar {:big? true
+                                         :locked? locked?
+                                         :emoji emoji
+                                         :emoji-background-color emoji-background-color}])]
+
+      [rn/view {:style (get styles :header-title-container)}
+       [text/text {:weight :semi-bold
+                   :number-of-lines 1
+                   :size  :heading-1
+                   :style (get styles :header-text)} (if (= type :community) name (str "# " name))]
+
+       (when (= type :community)
+         [icon/icon (if locked?
+                      :main-icons2/locked
+                      :main-icons2/unlocked)
+          {:style (get styles :header-title-lock)
+           :color           (if dark?
+                              colors/neutral-40
+                              colors/neutral-50)}])]
+
+      [button/button {:type :outline
+                      :size 32
+                      :icon true
+                      :style (get styles :header-info-button)} :info]]
```

Comment:
> `:info`

Hi, Sorry if I was not clear, but I didn't meant that we should use keyword instead of string. I meant we can put this string in `/translations/en.json` and then use it like `(i18n/label :t/info)`. So this string can be translated for other languages.

Reply:
I know, but i noticed that i was using icon buttons wrong, since i don't need to provide a label (string) at all 😃

Reply:
oh I see. So second variable/label is not optional? (maybe we can make it optional?)

![image](https://user-images.githubusercontent.com/17097240/193916180-e5252e65-c24c-4370-86a0-d47a2cf1c593.png)

Reply:
ok, I see now second variable in this case is icon. So don't we need to use it like `:main-icons2/icon`?

Reply:
if i understood it correctly, if `:icon true` is set, instead of a label the icon name is provided as the second argument

Reply:
Label would need to be provided, if there should be text inside the button

Reply:
yes probably, is adding `:main-icons2/...` to the icon name just a convention or is this technically necessary?

Reply:
I am not sure if it is necessary for quo2 button (maybe @flexsurfer will know better), but in general it is not just a convention. If you use `:main-icons/info` then file/image with `:info` name will be used as icon. But if you use `:main-icons2/info` then file/image with `:info+size` name will be used as icon. So in this case `:info32@2x.png` .([ref](https://github.com/status- im/status-mobile/blob/8dd75205f359d0f8b3cf02e5cf8e662812fae94f/src/quo2/components/icon.cljs#L10))

Reply:
namespace is not used, we cut it, its only for IDE, but quo2 button uses quo2 icons which adds sizes to the icon name

Reply:
Oh ok, got it. So technically adding the namespace doesn't change anything. The only difference is in using button component from either quo or quo2. Thanks!